File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        left = right = 0 
        ans = 0
        zeros = 0  
        while right < len(nums):
            if nums[right] == 0:
                zeros += 1 
            while zeros > 0:
                if nums[left] == 0: 
                    zeros -= 1
                left += 1
            ans = max(right-left+1,ans)
            right += 1
        return ans           

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int left = 0, right = 0;
    int numZeroes = 0, longestSeq = 0;

    while (right < nums.size()) {
        if (nums[right] == 0) {
            numZeroes++;
        }

        while (numZeroes > 0) {
            if (nums[left] == 0) {
                numZeroes--;
            }
            left++;
        }

        longestSeq = max(longestSeq, right - left + 1);
        right++;
    }

    return longestSeq;
        

    }
};